laituri/docker/credential_manager/ecr_with_role_v1.py

```diff
@@ -0,0 +1,48 @@
+import base64
+from contextlib import contextmanager
+from typing import Iterator
+
+from laituri.docker.credential_manager.docker_v1 import docker_v1_credential_manager
+from laituri.docker.credential_manager.errors import CallbackFailed, DockerLoginFailed
+from laituri.types import LogStatusCallable, RegistryCredentialsDict
+
+
+@contextmanager
+def ecr_with_role_v1_credential_manager(
+    *,
+    image: str,
+    registry_credentials: RegistryCredentialsDict,
+    log_status: LogStatusCallable,
+) -> Iterator[None]:
+    from boto3 import Session
+    from botocore.credentials import InstanceMetadataProvider
+    from botocore.utils import InstanceMetadataFetcher
+    fetcher = InstanceMetadataFetcher()
+    creds = fetcher.retrieve_iam_role_credentials()
+    if not creds['role_name'] == registry_credentials['role_name']:
+        # User tried to use a IAM role which is not attached to the EC2 machine
+        raise DockerLoginFailed(f"No instance role {registry_credentials['role_name']} found")
+
+    instance_metadata_provider = InstanceMetadataProvider(iam_role_fetcher=fetcher)
+    credentials = instance_metadata_provider.load().get_frozen_credentials()
```

Comment:
Don't you already have the `creds` dict above..? Why get it again (in a slightly different format)?